featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,264 @@
+    "cum_max",
+    "cum_min",
+    "cumulative_time_since_last_false",  # time_index
+    "cumulative_time_since_last_true",  # time_index
+    "diff",
+    "diff_datetime",
+    "exponential_weighted_average",
+    "exponential_weighted_std",
+    "exponential_weighted_variance",
+    "greater_than_previous",
+    "is_first_occurrence",
+    "is_last_occurrence",
+    "is_max_so_far",
+    "is_min_so_far",
+    "lag",  # time_index
+    "less_than_previous",
+    "percent_change",
+    "same_as_previous",
+    "time_since_previous",  # time_index
+]
+
+
+DEPRECATED_PRIMITIVES = [
+    # // as of premium - primitives 0.14.0 and feature - tools 0.26.1
+    "multiply_boolean",  # functionality duplicated by 'and' primitive
+    "numeric_lag",  # deperecated and replaced with `lag`
+]
+
+REQUIRED_INPUT_PRIMITIVES = [
+    # // as of premium - primitives 0.14.0 and feature - tools 0.26.1
+    "count_string",
+    "distance_to_holiday",
+    "is_in_geobox",
+    "score_percentile",
+    "subtract_numeric_scalar",
+    "scalar_subtract_numeric_feature",
+    "not_equal_scalar",
+    "multiply_numeric_scalar",
+    "modulo_numeric_scalar",
+    "divide_numeric_scalar",
+    "add_numeric_scalar",
+    "equal_scalar",
+    "greater_than_equal_to_scalar",
+    "less_than_equal_to_scalar",
+    "divide_by_feature",
+    "greater_than_scalar",
+    "less_than_scalar",
+    "modulo_by_feature",
+    "time_since",
+    "savgol_filter",
+    "isin",
+    "numeric_bin",
+]
+
+OTHER_PRIMITIVES_TO_EXCLUDE = [
+    "absolute",
+    "negate",
+    "not",
+    "and",
+    "is_zero",
+    "is_null",
+    "greater_than_equal_to",
+    "equal",
+    "greater_than",
+    "is_whole_number",
+    "not_equal",
+    "multiply_numeric",
+    "multiply_numeric_boolean",
+    "add_numeric",
+    "less_than_equal_to",
+    "divide_numeric",
+    "less_than",
+    "subtract_numeric",
+    "modulo_numeric",
+    "cosine",
+    "tangent",
+    "sine",
+]
+
+DEFAULT_EXCLUDED_PRIMITIVES = (
+    REQUIRED_INPUT_PRIMITIVES
+    + DEPRECATED_PRIMITIVES
+    + ORDERED_PRIMITIVES
+    + OTHER_PRIMITIVES_TO_EXCLUDE
+)
+
+TIME_SERIES_PRIMITIVES = [
+    "rolling_count",
+    "rolling_max",
+    "rolling_mean",
+    "rolling_min",
+    "rolling_std",
+    "rolling_trend",
+    "lag",
+]
+
+
+# TODO: Support multi-table
+def get_recommended_primitives(
+    entityset: EntitySet,
+    target_dataframe_name: str,
+    is_time_series: bool,
+    excluded_primitives: List[str] = DEFAULT_EXCLUDED_PRIMITIVES,
+) -> List[str]:
+    """Get a list of recommended primitives given an entity set.
+
+    Description:
+        Given a single table entity set with a `target_dataframe_name`, `is_time_series` specified
+        and a list of any primitives in `excluded_primitives` to not be included in the final recommendation list.
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `entityset`.
```

Comment:
I'm fine with that,  I figured I'd keep this in case we do support multi-table in the future. But if it's not looking like this will be done in the near future I can remove this parameter.

Reply:
I'd vote for removing it until we need it since we don't have near-term plans for multi-table.

Reply:
So yesterday, I did the former but just added some logic and exceptions if the number of DataFrames in the entityset is 0 or greater than 1. lmk what you think.